### ISSUE TITLE
Does not detect exports

### ISSUE THREAD

**Issue by gajus** [webpro/knip#119]
I have two workspaces.

`apps/foo` and `pkg/routines`

`pkg/routines` `package.json` defines:

```json
"exports": {
  "./*": {
    "import": "./dist/*.js",
    "types": "./dist/*.d.ts"
  }
},
```

* `src/bar.ts` is a file in `pkg/routines` that when compiled ends up in `dist/bar.js`
* `apps/foo` then imports `@pkg/routines/bar`

Unfortunately, Knip reports this file as unused:

```
Unused files (2)
pkg/routines/src/bar.ts
```

**Comment by webpro** [comment]
This is indeed a totally regular situation. Yet I think for Knip it isn't so straightforward:

- Assuming `dist` files are build artifacts, the file(s) might not even be there.
- The `dist` folder can be ignored using `.gitignore` (I suspect this is the culprit here).
- Not sure whether Knip should analyze build artifacts at all (reasons include performance, minified and/or modified code, importing other build artifacts, etc).

I expect that `pkg/routines` is referred to and marked as a used package from `apps/foo`.

So I expect `dist` is in `.gitignore` and I'm not sure whether that _should_ or even _could_ be dealt with. How can Knip know that  `dist/bar.js` is built from`src/bar.ts`?

**Comment by gajus** [comment]
... or perhaps I am missing something?

I don't see how (without this) can Knip function in a monorepo where the individual workspaces need to be built?

**Comment by webpro** [comment]
Another option is to add `src/bar.ts` as an entry file manually.

> ... or perhaps I am missing something?
> 
> I don't see how (without this) can Knip function in a monorepo where the individual workspaces need to be built?

Overall, the idea is more "where to even start fighting software rot" than "perfect outcome with absolutely no false positives". They're seen as bugs (and rightfully so!), but I see "gaps" :)

**Comment by gajus** [comment]
> This should already happen, but if I understand correctly the issue is that nothing imports src/bar.ts directly.

Ah, right. Yes, that's correct.

**Comment by webpro** [comment]
I've extended https://github.com/webpro/knip/blob/main/docs/handling-issues.md a bit

With that, and a related bug fix in v2.10.0 (related to handling ignored files in workspaces), I'm going to close this issue.

**Comment by eugenetree** [comment]
Just in case anybody is wondering, the link from the previous comment is mostly likely available here now https://knip.dev/guides/handling-issues
